foundation-model-stack/fms-hf-tuning
# Fix: Removal of transformers logger and addition of python native logger

### Description of the change

Aim was to move away from using logger modules from transformers and use python native [logger](https://docs.python.org/3/library/logging.html). Now, log level is either set by CLI argument of `--log_level` OR environment variable `LOG_LEVEL`. The setting of log level applies to both python native logger and log level of transformers logging.

### Related issue number

#[994](https://github.ibm.com/ai-foundation/watson-fm-stack-tracker/issues/994)

### How to verify the PR

- Set environment variable `LOG_LEVEL` as desired log level and check if logs are correct according to passed LOG_LEVEL, by running tuning job for a small model like `Maykeye/TinyLLama-v0`. 

Command:
```
python tuning/sft_trainer.py  \                                             
--model_name_or_path Maykeye/TinyLLama-v0  \
--training_data_path tests/data/twitter_complaints_small.json  \
--output_dir outputs/full-tuning  \
--num_train_epochs 5  \
--per_device_train_batch_size 2  \
--gradient_accumulation_steps 1  \
--learning_rate 1e-5  \
--response_template "\n### Label:"  \
--dataset_text_field "output" \
--use_flash_attn false \
--torch_dtype "float32"
```

- Pass `log_level` via cli argument, and check if logs are correct according to passed LOG_LEVEL, by running tuning job for a small model like `Maykeye/TinyLLama-v0`.

Command:
```
python tuning/sft_trainer.py  \                                             
--model_name_or_path Maykeye/TinyLLama-v0  \
--training_data_path tests/data/twitter_complaints_small.json  \
--output_dir outputs/full-tuning  \
--num_train_epochs 5  \
--per_device_train_batch_size 2  \
--gradient_accumulation_steps 1  \
--learning_rate 1e-5  \
--response_template "\n### Label:"  \
--dataset_text_field "output" \
--use_flash_attn false \
--torch_dtype "float32" \
--log_level "info"
```

### Was the PR tested
